File: RedSheet.DbEntities/Models/NanoOurBatna.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("NanoOurBatnas", Schema = "dbo")]
	[Serializable]
	public partial class NanoOurBatna  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoOurBatnaId { get; set; } 
 
	[Required] 
	[MaxLength(400)]
	public string NanoOurBatnaValue { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 

	public NanoOurBatna()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/nano-our-batna.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule,  vNanoOurBatnaRecord  } from './'
export class NanoOurBatna {
    constructor(nanoOurBatna?: NanoOurBatna  | vNanoOurBatnaRecord )  {
        let properties = [ "nanoOurBatnaId", "nanoOurBatnaValue", "projectModuleId",];
        for (let property of properties)
            if (nanoOurBatna && nanoOurBatna[property])
                this[property] = nanoOurBatna[property];
    }
 
	nanoOurBatnaId : number =   0 ;
 
    @required()
    @maxLength(400)
	nanoOurBatnaValue : string =   undefined;
 
    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;


}
